File: visitor/oc/bytedesk-oc/core/model/BDLeaveMsgModel.m
//
//  BDLeaveMsgModel.m
//  bytedesk-core
//
//  Created by 宁金鹏 on 2022/5/26.
//  Copyright © 2022 KeFuDaShi. All rights reserved.
//

#import "BDLeaveMsgModel.h"

@implementation BDLeaveMsgModel

-(instancetype)initWithDictionary:(NSDictionary *)dictionary {
    
    self = [super init];
    if (self) {
        _lid = [dictionary objectForKey:@"lid"];
        _avatar = [[dictionary objectForKey:@"visitor"] objectForKey:@"avatar"];
        _nickname = [[dictionary objectForKey:@"visitor"] objectForKey:@"nickname"];
        _client = [[dictionary objectForKey:@"visitor"] objectForKey:@"client"];
        _mobile = [dictionary objectForKey:@"mobile"];
        _email = [dictionary objectForKey:@"email"];
        _content = [dictionary objectForKey:@"content"];
        _createdAt = [dictionary objectForKey:@"createdAt"];
        _reply = [dictionary objectForKey:@"reply"];
        _replied = [[dictionary objectForKey:@"replied"] boolValue];
    }
    return self;
}

- (BOOL)isEqual:(id)object {
    BDLeaveMsgModel *msg = (BDLeaveMsgModel *)object;
    if ([_lid isEqualToString:msg.lid]) {
        return TRUE;
    }
    return FALSE;
}

@end

File: visitor/swift/Sources/bytedesk-swift/Models/BDLeaveMsgModel.swift
//
//  File.swift
//  
//
//  Created by 宁金鹏 on 2023/9/12.
//

import Foundation

public class BDLeaveMsgModel: Codable {
    
    var lid: String? // 唯一lid
    var avatar: String? // 留言人头像
    var nickname: String? // 留言人昵称
    var client: String? // 留言人来源
    var mobile: String? // 手机
    var email: String? // 邮箱
    var content: String? // 留言内容
    var createdAt: String? // 留言时间
    var reply: String? // 回复内容
    var replied: Bool // 是否已经回复
    
}
